packages/dataviews/src/types.ts

```diff
@@ -542,25 +543,45 @@ export type SimpleFormField = {
 	id: string;
 	layout?: 'regular' | 'panel';
 	labelPosition?: 'side' | 'top' | 'none';
-};
+} & { validation: FormFieldValidation };
 
 export type CombinedFormField = {
 	id: string;
 	label?: string;
 	layout?: 'regular' | 'panel';
 	labelPosition?: 'side' | 'top' | 'none';
 	children: Array< FormField | string >;
+} & { validation: FormFieldValidation };
+
+export type ValidationResult = {
+	isValid: boolean;
+	errorMessage: string | undefined;
 };
 
-export type FormField = SimpleFormField | CombinedFormField;
+export type FormFieldValidation = {
+	/**
+	 * The validation should be triggered only when the field is dirty.
+	 */
+	showErrorOnlyWhenDirty: boolean;
+	/**
+	 * The validation function.
+	 */
+	callback: ( data: any ) => ValidationResult;
+};
 
+export type FormField = SimpleFormField | CombinedFormField;
 /**
  * The form configuration.
  */
 export type Form = {
 	type?: 'regular' | 'panel';
 	fields?: Array< FormField | string >;
 	labelPosition?: 'side' | 'top' | 'none';
+	touchedFields: string[];
+	messageErrors: Record< string, string | undefined >;
+	setTouchedFields: ( touchedFields: string[] ) => void;
+	setErrors: ( field: string, error: string | undefined ) => void;
+	isFormValid: ( data: Record< string, any > ) => boolean;
```

Comment:
I think this is mixing several things together. `form` for me is about the "layout" of the form, how do we render the form. I think `validation state` should probably be separate from it. Maybe the current `form` prop name is what pushed you to add this stuff here and we might want to consider renaming it "view" or something like that.

Reply:
I don't have a strong opinion. IMHO form could be a big object with layout and validation option or spread them across multiple props.